moulinette/core.py

```diff
@@ -442,16 +442,21 @@ def acquire(self):
         start_time = time.time()
 
         while True:
-            lock_pid = self._lock_PID()
 
-            if lock_pid is None:
+            lock_pids = self._lock_PIDs()
+
+            if self._is_son_of(lock_pids):
+                return
+
+            if lock_pids == []:
                 self._lock()
                 break
             elif not self._stale_checked:
                 self._stale_checked = True
-                # Delete stale lock file
-                if not lock_pid or not os.path.exists(
-                        os.path.join('/proc', lock_pid, 'exe')):
+                # Check locked process still exist and take lock if it doesnt
+                # FIXME : what do in the context of multiple locks :|
```

Comment:
We might check `not os.path.exists(os.path.join('/proc', str(PID), 'exe'))` for each PID in the `_lock_PIDs` list ?